src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -84,6 +84,12 @@ public sealed class CompletionItem : IComparable<CompletionItem>
         /// </summary>
         public CompletionItemRules Rules { get; }
 
+        /// <summary>
+        /// Returns true if this item's text edit requires complex resolution that
+        /// may impact performance.
+        /// </summary>
+        public bool IsComplexTextEdit { get; }
```

Comment:
Consider an example, or stating something like: for example, if the edit may need to format or type check the resulting code, or make complex non-local changes to other parts of the file.